c10/core/DispatchKeySet.h

```diff
@@ -328,9 +677,11 @@ static inline DispatchKey legacyExtractDispatchKey(DispatchKeySet s) {
   // top of existing "backend" keys like CPU/CUDA, you need to add it
   // here.  At the moment, autograd keys and ADInplaceOrView key need this
   // treatment;
-  return (s - autograd_dispatch_keyset_with_ADInplaceOrView -
-          autocast_dispatch_keyset)
-      .highestPriorityTypeId();
+  auto ks = s.remove(DispatchKey::AutogradFunctionality)
```

Comment:
Why not keep using autograd_dispatch_keyset_with_ADInplaceOrView here ? If not, why are AutogradOther and AutogradNestedTensor not removed anymore?

Also, any hope legacyExtractDispatchKey could be removed altogether?

Reply:
you're right, I should replace this with `autograd_dispatch_keyset_with_ADInplaceOrView`. and I looked into where this function is called - it seemed a little more involved to try removing it in this PR, so I just tried to keep the same behavior for now.

Reply:
(and the individual autocast keys with their corresponding keyset)